guides/source/layouts_and_rendering.md

```diff
@@ -465,6 +465,10 @@ end
     end
 ```
 
+NOTE: Be stingy when adding new variants. Only reach for variants when groups
+of conditionals (that share a theme) or other more traditional templating
+techniques become unmanageable.
```

Comment:
This feels like an important caveat to make. There's a real dearth of information and conventions in the Rails ecosystem around variants, so I'm curious what the official stance should be. Please weigh in!

Reply:
My take is that unless Rails itself takes a stance, that is the result of an intentional decision by the team -- the rest is up to you on how to interpret that. But I think calling out trade-offs, or potential pitfalls, is a good way to put it without needing core to sign-off on a convention in the docs.

For example, _why_ is having too many variants a problem? How does it impact the application or developer?